Entities/Tower/Tower.cs

```diff
@@ -1,18 +1,118 @@
-using System;
+using System.Collections.Generic;
+using System.Linq;
+using ApproachTheForge.Effects;
 using ApproachTheForge.Utility;
 using Godot;
 
 namespace ApproachTheForge.Entities.Tower;
 
-public partial class Tower : Node2D, IPlaceable
+public partial class Tower : Node2D, IPlaceable, IDamageable
 {
+	[Export] public float Health { get; private set; } = 300f;
+
+	[ExportCategory("Damage Data")]
+	[Export] private float _damage = 40;
+	[Export] private float _knockback = 300;
+	[Export] private float _attackSpeed = 1f;
+	[Export] private float _attackDelay = 0.2f;
+
+	private PackedScene _projectileEffect = GD.Load<PackedScene>("res://Effects/projectile_effect.tscn");
+
+	private Area2D _detectionArea;
+	private GpuParticles2D _preShotParticle;
+	private AudioStreamPlayer2D _preShotAudio;
+	private float _timeSinceLastAttack;
+	private float _currentAttackDelay;
+	private Node2D _parent;
+	private bool _isDead;
+
+	private readonly List<Projectile> _activeProjectiles = new();
```

Comment:
I don't see a need to track the projectiles. Currently, they are just added then removed.

Reply:
I will in the future. If a projectile is out and the tower get's destroyed, the tower needs to stay partially alive long enough to process the last projectile.

Reply:
Kk